File: kotlin-dsl/src/test/java/com/chutneytesting/kolin/util/ChutneyServerInfoClearProperties.java
/*
 * SPDX-FileCopyrightText: 2017-2024 Enedis
 *
 * SPDX-License-Identifier: Apache-2.0
 *
 */

package com.chutneytesting.kolin.util;

import java.lang.annotation.ElementType;
import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import org.junitpioneer.jupiter.ClearSystemProperty;

@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.METHOD, ElementType.TYPE})
@Inherited
@ClearSystemProperty(key="http.proxyHost")
@ClearSystemProperty(key="http.proxyPort")
@ClearSystemProperty(key="http.proxyUser")
@ClearSystemProperty(key="http.proxyPassword")
@ClearSystemProperty(key="https.proxyHost")
@ClearSystemProperty(key="https.proxyPort")
@ClearSystemProperty(key="https.proxyUser")
@ClearSystemProperty(key="https.proxyPassword")
public @interface ChutneyServerInfoClearProperties {
}

File: kotlin-dsl/src/test/kotlin/util/ChutneyServerInfoClearProperties.kt
/*
 * SPDX-FileCopyrightText: 2017-2024 Enedis
 *
 * SPDX-License-Identifier: Apache-2.0
 *
 */

package util

import org.junitpioneer.jupiter.ClearSystemProperty
import java.lang.annotation.Inherited

@Target(AnnotationTarget.FUNCTION, AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@Inherited

@ClearSystemProperty(key="http.proxyHost")
@ClearSystemProperty(key="http.proxyPort")
@ClearSystemProperty(key="http.proxyUser")
@ClearSystemProperty(key="http.proxyPassword")
@ClearSystemProperty(key="https.proxyHost")
@ClearSystemProperty(key="https.proxyPort")
@ClearSystemProperty(key="https.proxyUser")
@ClearSystemProperty(key="https.proxyPassword")
annotation class ChutneyServerInfoClearProperties ()
